vspace/src/main/java/edu/asu/diging/vspace/web/staff/SlideController.java

```diff
@@ -36,20 +46,31 @@
 
     @Autowired
     private IContentBlockManager contentBlockManager;
+    
+    @Autowired
+    private SpaceRepository spaceRepo;
 
     @RequestMapping("/staff/module/{moduleId}/slide/{id}")
-    public String listSlides(@PathVariable("id") String id, @PathVariable("moduleId") String moduleId, Model model) {
+    public String listSlides(@PathVariable("id") String id, @PathVariable("moduleId") String moduleId, Model model) throws JsonProcessingException {
 
         ISlide slide = slideManager.getSlide(id);
+        List<ISpace> spaces = new ArrayList<ISpace>();
+        spaceRepo.findAll().forEach(s -> {
+            spaces.add(s);
+        });
+        model.addAttribute("spaces", spaces);
         model.addAttribute("module", moduleManager.getModule(moduleId));
         model.addAttribute("slide", slide);
         model.addAttribute("slideSequences", slideManager.getSlideSequences(id, moduleId));
         List<IContentBlock> slideContents = contentBlockManager.getAllContentBlocks(id);
+        HashMap<String, String> selectedSpaceForSpaceBlock = contentBlockManager.findSelectedSpaceForSpaceBlock(slideContents);
         model.addAttribute("slideContents", slideContents);
+        model.addAttribute("selectedSpaceForSpaceBlock", selectedSpaceForSpaceBlock);
```

Comment:
what are these needed for? each space block should have a space property with the linked space.

Reply:
Removed this, initially had included this to make the user know which is the current selected space for a space content block, but I have figured a way around this.